workspace/__ardulib__/Graphics/XRender.h

```diff
@@ -0,0 +1,26 @@
+
+#ifndef X_RENDERER_H
+#define X_RENDERER_H
+
+class XRenderer {
+public:
+    XRenderer() {}
+
+    virtual void setScanlinePart(int16_t scanline, int16_t xmin, int16_t xmax, uint16_t* lineBuffer) = 0;
+
+    int16_t getRendererWidth() const {
+        return _width;
+    }
+
+    int16_t getRendererHeight() const {
+        return _height;
+    }
+
+protected:
+
+    int16_t _width = 0;
+    int16_t _height = 0;

```

Comment:
Width and height are here, but there’s no way to set them. Perhaps, it is better to leave such details to specific implementations.